src/main/java/org/opensearch/knn/index/codec/transfer/OffHeapBinaryVectorTransfer.java

```diff
@@ -21,12 +23,19 @@ public OffHeapBinaryVectorTransfer(int transferLimit) {
 
     @Override
     public void deallocate() {
-        // TODO: deallocate the memory location
+        JNICommons.freeBinaryVectorData(getVectorAddress());
     }
 
     @Override
-    protected long transfer(List<byte[]> vectorsToTransfer, boolean append) throws IOException {
-        // TODO: call to JNIService to transfer vector
-        return 0L;
+    protected long transfer(List<byte[]> batch, boolean append) throws IOException {
+        if (!batch.isEmpty()) {
+            return JNICommons.storeBinaryVectorData(
+                getVectorAddress(),
+                batch.toArray(new byte[][] {}),
+                (long) batch.get(0).length * transferLimit,
+                append
+            );
+        }
+        return 0;
```

Comment:
Should this return the vector address? Just thinking in append scenario

Reply:
Actually, we don't need to check if list is empty or not because we are calling this if list size is equal to transfer limit or during flush (where we are verifying if list is not empty)